; gtk_main executes GDK_THREADS_LEAVE()
; which causes gdk to unlock a thread which was only created but
; never locked.
; According glib's source, unlocking a not-locked thread is
; undefined behavior, which on Linux seems to be "nothing", but
; on FreeBSD is EPERM, and glib thus aborts, and this is not
; distinguishable from an attempt to unlock another thread's mutex
; thus we hack away these calls.
; THIS IS A HACK - it seems to work, but this problem should be
; taken upstream and fixed properly.
Index: gtk/gtkmain.c
===================================================================
--- gtk/gtkmain.crig	2013-02-23 12:52:48.506521649 +0100
+++ gtk/gtkmain.c	2013-02-23 12:52:51.522521750 +0100
@@ -1253,9 +1253,9 @@ gtk_main (void)
 
   if (g_main_loop_is_running (main_loops->data))
     {
-      GDK_THREADS_LEAVE ();
+      //GDK_THREADS_LEAVE ();
       g_main_loop_run (loop);
-      GDK_THREADS_ENTER ();
+      //GDK_THREADS_ENTER ();
       gdk_flush ();
     }
 
